src/main/java/com/hermesworld/ais/galapagos/topics/service/impl/MessagesService.java

```diff
@@ -0,0 +1,54 @@
+package com.hermesworld.ais.galapagos.topics.service.impl;
+
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+import org.springframework.web.servlet.support.RequestContextUtils;
+
+import java.text.MessageFormat;
+import java.util.Locale;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+
+@Slf4j
+public class MessagesService {
+
+    private final String packageName;
+
+    public MessagesService(Class<?> clazz) {
+        packageName = clazz.getPackage().getName() + ".messages";
+    }
+
+    public String getMessage(String key, Object... args) {
+
+        Locale locale = getLocale();
+        ResourceBundle rb = ResourceBundle.getBundle(packageName, locale);
+        String message;
+        try {
+            message = rb.getString(key);
+        }
+        catch (MissingResourceException e) {
+            log.warn("No matching message for the provided key {}", key);
+            return key;
+        }
+
+        return MessageFormat.format(message, args);
+    }
+
+    public static Locale getLocale() {
```

Comment:
Please move this method to a utility class named `MessageUtil` in package `com.hermesworld.ais.galapagos.messages`, and rename the method to `getCurrentLocale()`.